### PR TITLE
mgr/dashboard: increase timeout to detect replication user in the secondary cluster

### PR DESCRIPTION
**Pull request by aaSharma14**
Increase timeout to detect replication user in the secondary cluster in rgw multisite automation wizard. Currently its set to 2 mins, increase it to 5 minutes.
When you import realm token to the secondary cluster, we wait for the replication/system user we created in the primary cluster to be present in the secondary cluster and when we find that user we set the credentials in the secondary cluster using ceph dashboard set-rgw-crdentials . The timeout for this is set to 2 minutes and sometimes it takes more than 2 minutes for the user to be replicated in the secondary cluster

Fixes: https://tracker.ceph.com/issues/68384

Signed-off-by: Aashish Sharma <aasharma@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by afreen23 — COMMENTED** [review]
Hey , why we need to increase the timeout ?

**Review by Pegonzal — COMMENTED** [review]
Hey @aaSharma14 I was under the impression we were going to start using a maximum number of tries instead of using a timeout

**Comment by aaSharma14** [comment]
> Hey , why we need to increase the timeout ?

when you import realm token to the secondary cluster, we wait for the replication/system user we created in the primary cluster to be present in the secondary cluster and when we find that user we set the credentials in the secondary cluster using ceph dashboard set-rgw-crdentials  . The timeout for this is set to 2 minutes and sometimes it takes more than 2 minutes for the user to be replicated in the secondary cluster

**Comment by aaSharma14** [comment]
> Hey @aaSharma14 I was under the impression we were going to start using a maximum number of tries instead of using a timeout

As discussed in the standup, we decided to increase the timeout for now to 5 minutes

**Comment by afreen23** [comment]
> > Hey , why we need to increase the timeout ?
> 
> when you import realm token to the secondary cluster, we wait for the replication/system user we created in the primary cluster to be present in the secondary cluster and when we find that user we set the credentials in the secondary cluster using ceph dashboard set-rgw-crdentials . The timeout for this is set to 2 minutes and sometimes it takes more than 2 minutes for the user to be replicated in the secondary cluster

You see how beautiful it becomes when we get to know reasons  behind.
Please add them in your commits and PRs and get faster reviews.

Thanks

**Review by afreen23 — APPROVED** [review]

File: src/pybind/mgr/dashboard/services/rgw_client.py [hunk 1]
```diff
@@ -1298,7 +1298,7 @@ def check_user_in_second_cluster(self, cluster_url, cluster_token, username,
         user_found = False
         start_time = time.time()
         while not user_found:
-            if time.time() - start_time > 120:  # Timeout after 2 minutes
+            if time.time() - start_time > 300:  # Timeout after 5 minutes
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/services/rgw_client.py:1301]: nit: Later on refactor to make it a variable and use variable in condition.

**Review by cloudbehl — APPROVED** [review]

**Comment by aaSharma14** [comment]
jenkins test make check

**Comment by aaSharma14** [comment]
jenkins test make check

**Comment by aaSharma14** [comment]
jenkins test make check

**Pull request merged**